CMakeLists.txt

```diff
@@ -0,0 +1,277 @@
+
+set(PACKAGE_VERSION "${PROJECT_VERSION}")
+set(VERSION "${PROJECT_VERSION}")
+configure_file(${CMAKE_CURRENT_LIST_DIR}/mbus/config.h.in ${CMAKE_CURRENT_LIST_DIR}/config.h @ONLY)
+
+add_library(
+  ${PROJECT_NAME}
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-protocol.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-protocol.h
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-tcp.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-tcp.h
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus.h
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-protocol-aux.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-protocol-aux.h
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-serial.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-serial.h)
+target_include_directories(
+  ${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}> $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+                         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "Android")
+  target_link_libraries(${PROJECT_NAME} PRIVATE m)
+endif()
+if(NOT MSVC)
+  target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra -Wno-pedantic)
+endif()
+
+if(CLANG_TIDY_EXE)
+  set_target_properties(${PROJECT_NAME} PROPERTIES CXX_CLANG_TIDY "${DO_CLANG_TIDY}")
+endif()
+
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+
+# ##############################################################################
+# examples
+# ##############################################################################
+
+if(LIBMBUS_BUILD_EXAMPLES)
+  message(STATUS "building examples")
+  add_subdirectory(bin)
+endif()
+
+# ##############################################################################
+# tests
+# ##############################################################################
+
+if(LIBMBUS_BUILD_TESTS)
+  message(STATUS "building tests")
+  enable_testing()
+  add_subdirectory(test)
+endif()
+
+# ##############################################################################
+# install
+# ##############################################################################
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
+set(INSTALL_PKGCONFIG_DIR
+    "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig"
+    CACHE PATH "Installation directory for pkgconfig (.pc) files")
+set(INSTALL_INC_DIR
+    "${CMAKE_INSTALL_INCLUDEDIR}/mbus"
+    CACHE PATH "Installation directory for headers")
+set(INSTALL_LIB_DIR
+    "${CMAKE_INSTALL_LIBDIR}"
+    CACHE PATH "Installation directory for libraries")
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libmbus.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libmbus.pc @ONLY)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libmbus.pc DESTINATION "${INSTALL_PKGCONFIG_DIR}")
+
+set(LIBMBUS_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+install(
+  TARGETS ${PROJECT_NAME}
+  EXPORT ${PROJECT_NAME}Targets
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT lib
```

Comment:
```suggestion
  LIBRARY DESTINATION ${INSTALL_LIB_DIR} COMPONENT lib
  ARCHIVE DESTINATION ${INSTALL_LIB_DIR} COMPONENT lib
```

Reply:
whats the difference?

Reply:
I had to look it up myself.
See https://cmake.org/cmake/help/latest/command/install.html#installing-targets

Archive: *.a, *.lib
Library: *.so, *.dylib
Runtime: *.dll, *.exe